tests/integrations/raftstore/test_replication_mode.rs

```diff
@@ -456,6 +468,82 @@ fn test_migrate_replication_mode() {
     assert_eq!(state.state, RegionReplicationState::IntegrityOverLabel);
 }
 
+#[test]
+fn test_migrate_majority_to_drautosync() {
+    // 1. start cluster, enable dr-auto-sync and set labels.
+    let mut cluster = new_server_cluster(0, 3);
+    cluster.pd_client.disable_default_operator();
+    cluster.cfg.raft_store.pd_store_heartbeat_tick_interval = ReadableDuration::millis(50);
+    cluster.cfg.raft_store.raft_log_gc_threshold = 10;
+    prepare_labels(&mut cluster);
+    cluster.run();
+    cluster.must_transfer_leader(1, new_peer(1, 1));
+    cluster.must_put(b"k1", b"v0");
+    cluster.pd_client.configure_dr_auto_sync("dc");
+    thread::sleep(Duration::from_millis(100));
+    let region = cluster.get_region(b"k1");
+    let mut request = new_request(
+        region.get_id(),
+        region.get_region_epoch().clone(),
+        vec![new_put_cf_cmd("default", b"k2", b"v2")],
+        false,
+    );
+    request.mut_header().set_peer(new_peer(1, 1));
+    let (cb, mut rx) = make_cb(&request);
+    cluster
+        .sim
+        .rl()
+        .async_command_on_node(1, request, cb)
+        .unwrap();
+    assert_eq!(rx.recv_timeout(Duration::from_millis(100)).is_ok(), true);
+    must_get_equal(&cluster.get_engine(1), b"k2", b"v2");
+    let state = cluster.pd_client.region_replication_status(region.get_id());
+    assert_eq!(state.state_id, 1);
+    assert_eq!(state.state, RegionReplicationState::IntegrityOverLabel);
+
+    // 2. swith to marjority mode.
+    cluster
+        .pd_client
+        .switch_replication_mode(DrAutoSyncState::Async, vec![]);
+    thread::sleep(Duration::from_millis(150));
+
+    // 3. spilt the region and make a new region, the regions status must be
```

Comment:
current DR will forbid the split during the Async stage. Is there any other situation causing this problem?

Reply:
um...? it seems to split successfully during the async stage.
I printed the detail of the region_m and region_k.
region_m: id: 1 start_key: 6D31 region_epoch { conf_ver: 1 version: 2 } peers { id: 1 store_id: 1 } peers { id: 2 store_id: 2 } peers { id: 3 store_id: 3 }, 
region_k: id: 1000 end_key: 6D31 region_epoch { conf_ver: 1 version: 2 } peers { id: 1001 store_id: 1 } peers { id: 1002 store_id: 2 } peers { id: 1003 store_id: 3 }

Reply:
split will be forbidden on the pd side, so your test can split with mock pd.